src/dune_rules/inline_tests.ml

```diff
@@ -343,35 +343,40 @@ include Sub_system.Register_end_point (struct
           | Native | Best | Byte -> Memo.return Alias0.runtest
           | Jsoo mode -> Jsoo_rules.js_of_ocaml_runtest_alias ~dir ~mode
         in
-        Super_context.add_alias_action
-          sctx
-          ~dir
-          ~loc:info.loc
-          (Alias.make ~dir runtest_alias)
-          (let open Action_builder.O in
-           let+ actions =
-             let* partitions_flags =
-               match partitions_flags with
-               | None -> Action_builder.return [ None ]
-               | Some _ ->
-                 let+ partitions = Action_builder.lines_of (Path.build partition_file) in
-                 List.map ~f:(fun x -> Some x) partitions
-             in
-             List.map partitions_flags ~f:(fun p -> action mode (flags p))
-             |> Action_builder.all
-           and+ () = Action_builder.paths source_files in
-           match actions with
-           | [] -> Action.Full.empty
-           | _ :: _ ->
-             let run_tests = Action.concurrent actions in
-             let diffs =
-               List.map source_files ~f:(fun fn ->
-                 Path.as_in_build_dir_exn fn
-                 |> Path.Build.extend_basename ~suffix:".corrected"
-                 |> Promote.Diff_action.diff ~optional:true fn)
-               |> Action.concurrent
-             in
-             Action.Full.make ~sandbox @@ Action.progn [ run_tests; diffs ]))
+        Memo.parallel_iter
+          [ runtest_alias; Alias.Name.of_string (Lib_name.Local.to_string lib_name) ]
```

Comment:
Can you make runtest depend on the <lib_name> alias instead? Aren't we running the tests twice if building both @runtest and @<lib_name> aliases?

Reply:
I don't think this is how aliases work. But maybe @rgrinberg can confirm. IIUC the aliases should point to the same target, so if they are both requested in the engine, only that target would end up getting built, and nothing will be run twice. But it's possible I have misunderstood the situation.

Reply:
Hugo's right, we are going to run the tests twice this way. You should create the alias and then add it as a dependency to runtest. A test to demonstrate this would be valuable as well btw.

> IIUC the aliases should point to the same target, so if they are both requested in the engine, only that target would end up getting built, and nothing will be run twice

What I think you you forgot is that the anonymous action we attach to the aliases will have a different digest. The digest of the anonymous action here will include the alias name, and therefore the action will be re-executed.

Reply:
It seems we are doing something similar in `Cram_rules.test_rule`. There we iterate over all the aliases and do `Super_context.add_alias_action` like I do here. I don't ever recall `@runtest` and `@mycramtest` building things twice.